### PR TITLE
xcaddy: 0.3.1 -> 0.3.2, add indeednotjames  as maintainer

### PR DESCRIPTION
**Pull request by emilylange**
###### Description of changes

upstream: https://github.com/caddyserver/xcaddy/releases/tag/v0.3.2

nixpkgs: Patches the version reporting when running `xcaddy version` to return the proper version, instead of go's default "(devel)"

note: the `vendorHash` did not change, because the project's `go.sum` didn't change either

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by winterqt — CHANGES_REQUESTED** [review]
Can you split this into separate commits for each change?

**Review by winterqt — CHANGES_REQUESTED** [review]
The maintainer addition is also a distinct change, please split that, too :)

(`xcaddy: add indeednotjames as maintainer` works if you can't think of a better commit message.)

Also, can you append to the list instead of adding yourself (assumingely) alphabetically? I believe that's more common.

(I'm trying to collect data on this, to infer future potential guidelines. If I may ask, (a) why did you initially not split these up, and (b) why did you not split/think of the maintainer addition as a separate change? Thank you!)

**Comment by IndeedNotJames** [comment]
> (a) why did you initially not split these up

As for the `xcaddy version` fix, I think because I remembered other PRs fixing build info/reporting (at least in golang) not splitting it either.
E.g. caddy pre-#192351 reported `(devel)` (and later an empty string) as well iirc.
Its version info patch wasn't really specific to `2.6.0 -> 2.6.1` either.

And I think I came across multiple other similar cases as well.
So I tried to copy what others did, I guess.

But I agree with you that it makes sense to split them!

---

> (b) why did you not split/think of the maintainer addition as a separate change? Thank you!

Similar to (a):
I remember other maintainers added themselves in a single commit along with the actual change as well.
Something that could be split into two (since they aren't logically connected), but it seems to be a very common thing to just sweep under the rug (or in this case, commit).

But again, I agree with you that it makes sense to split them :)

---

> Also, can you append to the list instead of adding yourself (assumingely) alphabetically? I believe that's more common.

Alphabetically, yes.
Did change it now, but I thought that would be more common O.o

File: pkgs/servers/caddy/xcaddy/default.nix
Comment (winterqt) [inline pkgs/servers/caddy/xcaddy/default.nix:23]: *Technically* this commit does more than fix the build info (strips symbols and DWARF info), but I feel like I'm being too much of a pedant for things we don't have concrete guidelines for, so it's not worth changing -- just figured I'd note it.

**Comment by winterqt** [comment]
Thank you for the detailed response, more or less boils down to what I thought it would (as in, it's what you've seen others do).

> Did change it now, but I thought that would be more common O.o

[Looks like it wildly varies](https://github.com/search?q=repo%3ANixOS%2Fnixpkgs+%22as+maintainer%22&type=commits), so maybe I was being a bit too pedantic there.

With regards to the commit granularity: I believe it's not ideal, and I'm trying to change that by writing some guidelines we can iterate on and enforce etc., as the Git history right now is a bit of a mess. I appreciate you taking the time to give your perspective, thanks again! :)

**Review by winterqt — APPROVED** [review]
There's two nits that I won't have you go back and change (because it feels somewhat cruel due to the whole no guidelines thing), but we typically include links to changelogs in bumps.

(waiting on @tjni's approval before merging)

**Comment by IndeedNotJames** [comment]
> we typically include links to changelogs in bumps

I linked to the changelog in my original PR body:
 
> upstream: https://github.com/caddyserver/xcaddy/releases/tag/v0.3.2

(that release description has an auto-generated section called *Changelog*)

I just prefixed it with *upstream:* instead of *changelog:*, because I wanted to differentiate between *changelog* (as in summary, including nixpkgs-patch-stuff) and *upstream changelog* 

> upstream: https://github.com/caddyserver/xcaddy/releases/tag/v0.3.2
> 
> nixpkgs: Patches the version reporting when running `xcaddy version` to return the proper version, instead of go's default "(devel)"
> 
> note: the `vendorHash` did not change, because the project's `go.sum` didn't change either

Or did you mean the commit body? :eyes:

**Comment by SuperSandro2000** [comment]
IMO from my interpretation of the contributing guide we don't need to split each change into such a small commit especially for such small, uncontroversial and straight forward changes.
Also regarding that we would need to explain this to a lot of people and it causes a lot back and forth.

**Comment by winterqt** [comment]
@IndeedNotJames I meant the commit body.

@SuperSandro2000 I was waiting on the other maintainer before merging, why didn't you?

**Comment by IndeedNotJames** [comment]
@SuperSandro2000 why did you remove the backport label?

That version bump isn't a breaking change.

**Comment by winterqt** [comment]
> IMO from my interpretation of the contributing guide we don't need to split each change into such a small commit especially for such small, uncontroversial and straight forward changes.
Also regarding that we would need to explain this to a lot of people and it causes a lot back and forth.

Hard disagree. We're a large enough project and things like confusing commits that don't focus on what they state they do are not acceptable, IMO.

**Comment by winterqt** [comment]
@tjni Think we should backport this? I'm indifferent -- don't see why not, but figured I'd ask your opinion on it as well.

**Comment by SuperSandro2000** [comment]
> why did you remove the backport label?
> 
> That version bump isn't a breaking change.

We usually only backport bug and security fixes, not regular patch updates. I didn't find anything in the PR body that justified the backport so I removed the backport label as usual.

So the question to ask is ``Why backport it?`` and not ``Why not backport it?``. If you can answer that then feel free to add the label back.

> Hard disagree. We're a large enough project and things like confusing commits that don't focus on what they state they do are not acceptable, IMO.

We don't need to be to strict about commit messages. If people split it then that's fine, if not and the changes are small enough, then that's fine, too. I don't expect people to have the time and motivation to fix their commit messages multiple times unless they are really not that great or completely wrong.

> I was waiting on the other maintainer before merging, why didn't you?

Mostly because the changes are uncontroversial and I didn't want to leave the author in limbo state with their PR.

**Comment by winterqt** [comment]
> We usually only backport bug and security fixes, not regular patch updates. I didn't find anything in the PR body that justified the backport so I removed the backport label as usual.

That's *just not true*. We backport more or less anything that wants to be backported that isn't breaking.

> We don't need to be to strict about commit messages. If people split it then that's fine, if not and the changes are small enough, then that's fine, too. I don't expect people to have the time and motivation to fix their commit messages multiple times unless they are really not that great or completely wrong.

I will simply restate my previous statement, as it is still true:

Hard disagree. We're a large enough project and things like confusing commits that don't focus on what they state they do are not acceptable, IMO.

Oh, and also: every time I've asked people to fix their commits, they not only do it, but also mostly agree with me! 

> Mostly because the changes are uncontroversial and I didn't want to leave the author in limbo state with their PR.

Doesn't mean you have to go and merge something that I touched not even hours before, but you already apologized for this, so shrug.

**Pull request merged**
